File: src/main/java/org/apache/prepbuddy/smoothers/KeyPartitioner.java
package org.apache.prepbuddy.smoothers;

import org.apache.spark.Partitioner;

/**
 * A Patitioner that puts each row in a partition that we specify by key.
 */
public class KeyPartitioner extends Partitioner {
    private int numPartitions;

    public KeyPartitioner(int numPartitions) {
        this.numPartitions = numPartitions;
    }

    @Override
    public int numPartitions() {
        return numPartitions;
    }

    @Override
    public int getPartition(Object key) {
        return (int) key;
    }
}

File: src/main/scala/org/apache/datacommons/prepbuddy/smoothers/KeyPartitioner.scala
package org.apache.datacommons.prepbuddy.smoothers

import org.apache.spark.Partitioner

/**
  * A Patitioner that puts each row in a partition that we specify by key.
  */
class KeyPartitioner(numOfPartitions: Int) extends Partitioner{
    override def numPartitions: Int = numOfPartitions

    override def getPartition(key: Any): Int = key.toString.toInt
}
